jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -73,13 +75,38 @@ public boolean isCanceled() {
 			for (String uri : uris) {
 
 				URI u = JDTUtils.toURI(uri);
+
 				ICompilationUnit unit = JDTUtils.resolveCompilationUnit(u);
 				if (unit != null) {
 					// System.out.println("--class name: " + unit.getElementName());
 					IType[] alltypes;
+					IAnnotation[] allAnnotations;
 					try {
 						alltypes = unit.getAllTypes();
 						for (IType type : alltypes) {
+							allAnnotations = type.getAnnotations();
+							boolean isWebServletAnnotated = false;
+							boolean isHttpServletExtended = false;
+							for (IAnnotation annotation : allAnnotations) {
+								if (annotation.getElementName() == "WebServlet") {
+//									System.out.println("--Annotation name: " + annotation.getElementName());
+									isWebServletAnnotated = true;
+								}
+							}
+
+							String typeExtension = type.getSuperclassTypeSignature();
```

Comment:
If you use `type.getSuperclassName()`, you will receive the name instead of the typeSignature (no leading `Q` in the name)